### DISCUSSION TITLE
PyVista <==> xarray

### DISCUSSION THREAD

**Discussion by RichardScottOZ** [pyvista/pyvista#2375] [Ideas]
Two things I use more and more it seems.

There's a discussions here:-

https://github.com/pydata/xarray/issues/4470#issuecomment-1068428147

e.g. @banesullivan did an extremely useful https://banesullivan.com/pyvista/examples/raster.html#sphx-glr-pyvista-examples-raster-py [and an earlier version, but I am a big fan of rioxarray too]

and things I have literally done recently:-

![image](https://user-images.githubusercontent.com/72196131/160086222-c347913f-969f-4e35-8ae9-aabc9cb8e304.png)

to xarray

```python
data = result['density'].reshape(result.dimensions[2],result.dimensions[1],result.dimensions[0])
XC = [result.bounds[0]+200*i for i in range(result.dimensions[0])]
YC = [result.bounds[2]+200*i for i in range(result.dimensions[1])]
ZC = [result.bounds[4]+200*i for i in range(result.dimensions[2])]
da = xr.DataArray(data=data, dims=["z", "y","x"], coords={"z": ZC, "y": YC, "x": XC})
```

![image](https://user-images.githubusercontent.com/72196131/160086715-f61f78b1-f3ad-497d-bf6f-85db63932921.png)

as a quick version

so having this officially integrated where you didn't have to think about it for various types of things would be excellent.

So where would you start?

A wrap thing and accessors on the other side, or all in that?

I don't have a good idea of where I would start, yet, PyVista developers might though.

**Comment by RichardScottOZ** [comment]
I will note that the grid happily came out of a PVGeo UBC Mesh reader - so thanks for that, too.

**Comment by banesullivan** [comment]
FYI, this effort has been implemented in pyvista-xarray, please head there for any further feedback, usage, thoughts, etc.

https://github.com/pyvista/pyvista-xarray

**Reply by RichardScottOZ** [reply]
Yes, working nicely.
